src/main/java/com/ibm/watson/modelmesh/ModelMeshApi.java

```diff
@@ -716,12 +727,24 @@ public void onHalfClose() {
                                         balancedMetaVal, headers, reqMessage, allRequired);
                             }
                         } finally {
+                            try {
+                                payloadProcessor.processRequest(new Payload(payloadId, modelId, String.valueOf(isVModel),
+                                                                                     methodName, headers, reqMessage.retainedDuplicate()));
+                            } catch (Throwable t) {
+                                logger.warn("Error while processing request payload {}", t.getMessage());
+                            }
                             releaseReqMessage();
                         }
 
                         respSize = response.data.readableBytes();
                         call.sendHeaders(response.metadata);
                         call.sendMessage(response.data);
+                        try {
+                            payloadProcessor.processResponse(new Payload(payloadId, modelId, String.valueOf(isVModel), methodName,
+                                                                         response.metadata, response.data.retainedDuplicate()));
```

Comment:
I don't _think_ we should need to call duplicate on the bytbufs since they are "finished with" in both cases. It may be that we need to reset the reader/writer indices (and possibly record the buffer's current indices before the response is actually sent to allow for this).

Reply:
ok, the read and write indexes are recorded for both request and responses and reset for the sake of payload processing.

Reply:
however it seems that a `retainDuplicate` is anyway required to avoid `ByteBufs` being GC'ed before the async processor kicks in

Reply:
> however it seems that a `retainDuplicate` is anyway required to avoid `ByteBufs` being GC'ed before the async processor kicks in

This won't be a problem if we do as suggested [above](https://github.com/kserve/modelmesh/pull/84#discussion_r1125734035).

Reply:
ok, resolving the above made this `retainDuplicate` not necessary anymore.